0000-zk-framework.md

```diff
@@ -0,0 +1,117 @@
+backend.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This project demonstrates R1CS logic with backends to Bellman and Bulletproofs.
+
+Use the following commands to execute each step of the process. The steps are setup (bellman),
+prove, and verify.
+
+Setup
+
+`cargo run -- setup --backend=bellman | tee params.json`
+
+Prove
+
+`echo '"1234"' > witness.hex`
+`cargo run -- prove --backend=bellman --parameters=params.json --witness=witness.hex | tee
+proof.json`
+
+or 
+
+`cargo run -- prove --backend=bulletproofs --witness=witness.hex | tee proof.json`
+
+Verify
+
+`cargo run -- verify --backend=bellman --parameters=params.json --input=proof.json`
+
+or
+
+`cargo run -- verify --backend=bulletproofs --input=proof.json`
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+An R1CS instance can be defined by three matrices A, B and C. They correspond to the NP-complete
+decision problem: whether there is a witness vector w s.t. Aw x Bw = Cw ?
+
+main.rs: this file contains sample code on how to use the framework, get_circuit function defines
+the circuit for which zero knowledge proofs will be generated
+r1csbellman.rs, r1csbulletproof.rs: modules that convert r1cs spec to underlying backends
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+We have chosen r1cs rust library (https://github.com/mir-protocol/r1cs) as frontend, which might
+not be the preferred frontend for all users.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not
+  choosing them?
+- What is the impact of not doing this?
+- For incorporating new protocol implementations what other implementations
+  exist and why were they not selected?
+- For new protocols, what related protocols exist and why do the not satisfy
+  requirements?
+
+One alternative is to use zk-interface (mentioned in prior arts). However, in our
+experience setting up zk-interface can be difficult. Other alternative can be directly using the
+underlying Zero Knowledge libraries such as Bellman, Bulletproof, etc. However, that would make the
+development process less flexible.
+
+# Prior art
+[prior-art]: #prior-art
+
+zk-interface (https://github.com/QED-it/zkinterface) is the defacto prior art. It has extensive
+features. However, it is not being actively maintained and set up is relatively complex.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the RFC process
+  before this gets merged?
+- What parts of the design do you expect to resolve through the implementation
+  of this feature before stabilization?
+- What related issues do you consider out of scope for this RFC that could be
+  addressed in the future independently of the solution that comes out of this
+  RFC?
```

Comment:
same comments here as in the alternatives section. The text is okay, but doesn't directly answer the questions.
It is also a good place to talk about all the things this RFC doesn't do (and why).